### PR TITLE
Wrap eth

### PR DESCRIPTION
**Pull request by rudewalt**

### CODE REVIEW

File: packages/contracts/contracts/MarginlyPoolWrapper.sol [hunk 1]
```diff
@@ -43,35 +47,87 @@ contract MarginlyPoolWrapper is IMarginlyPoolWrapper {
     _;
   }
 
+  receive() external payable {
+    require(false); //Nobody can send ETH without calldata
```
Comment (Overseven) [inline packages/contracts/contracts/MarginlyPoolWrapper.sol:51]: Receive method description:
https://docs.soliditylang.org/en/v0.8.12/contracts.html#receive-ether-function

WETH code:
https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#code

```
function withdraw(uint wad) public {
  ...
  msg.sender.transfer(wad);
  ...
}
```

File: packages/contracts/contracts/MarginlyPoolWrapper.sol [hunk 2]
```diff
@@ -2,24 +2,28 @@
...
-    for(uint256 index = 0; index < marginlyPoolAddresses.length; ++index) {
+  address public immutable WETH9;
+
+  constructor(address[] memory marginlyPoolAddresses, address _owner, address _WETH9) {
+    for (uint256 index = 0; index < marginlyPoolAddresses.length; ++index) {
       whitelistedMarginlyPools[marginlyPoolAddresses[index]] = true;
     }
     unlocked = true;
     owner = _owner;
+    WETH9 = _WETH9;
```
Comment (Overseven) [inline packages/contracts/contracts/MarginlyPoolWrapper.sol:26]:
```
require(_WETH9 != address(0));
```

File: packages/contracts/contracts/MarginlyPoolWrapper.sol [hunk 3]
```diff
@@ -2,24 +2,28 @@
...
   /// @dev contract admin
   address public owner;
 
-  constructor(address[] memory marginlyPoolAddresses, address _owner) {
-    for(uint256 index = 0; index < marginlyPoolAddresses.length; ++index) {
+  address public immutable WETH9;
+
+  constructor(address[] memory marginlyPoolAddresses, address _owner, address _WETH9) {
+    for (uint256 index = 0; index < marginlyPoolAddresses.length; ++index) {
       whitelistedMarginlyPools[marginlyPoolAddresses[index]] = true;
```
Comment (Overseven) [inline packages/contracts/contracts/MarginlyPoolWrapper.sol:22]: check marginlyPools existing is required

File: packages/contracts/contracts/MarginlyPoolWrapper.sol [hunk 4]
```diff
@@ -43,35 +47,87 @@ contract MarginlyPoolWrapper is IMarginlyPoolWrapper {
...
+  /// @dev Internal version without lock modifier
+  function depositQuoteInternal(IMarginlyPool marginlyPool, uint256 amount) internal {
+    address quoteToken = marginlyPool.quoteToken();
+    wrapEthOrTransferErc20(quoteToken, amount);
+    TransferHelper.safeApprove(quoteToken, address(marginlyPool), amount);
+    marginlyPool.depositQuote(amount);
+  }
+
+  function wrapEthOrTransferErc20(address token, uint256 amount) internal {
+    if (token == WETH9 && address(this).balance >= amount) {
```
Comment (Overseven) [inline packages/contracts/contracts/MarginlyPoolWrapper.sol:113]:
```
require(token != address(0));
```

Comment (rudewalt) [reply packages/contracts/contracts/MarginlyPoolWrapper.sol:113]: It is an internal method, token addresses come from the pool and couldn't be address(0)

**Pull request closed without merging**